alibi_detect/saving/loading.py

```diff
@@ -216,14 +215,20 @@ def _load_preprocess_config(cfg: dict,
             emb = kwargs.pop('embedding')  # embedding passed to preprocess_drift as `model` therefore remove
 
             # Backend specifics
-            if backend == 'tensorflow':
+            if has_tensorflow and isinstance(model, SupportedModels_tf):
                 model = prep_model_and_emb_tf(model, emb)
                 kwargs.pop('device')
-            elif backend == 'pytorch':  # TODO - once optional deps implemented
+            elif has_pytorch and isinstance(model, SupportedModels_torch):
                 raise NotImplementedError('Loading preprocess_fn for PyTorch not yet supported.')
 #               device = cfg['device'] # TODO - device should be set already - check
 #               kwargs.update({'model': kwargs['model'].to(device)})  # TODO - need .to(device) here?
 #               kwargs.update({'device': device})
+            elif model is None:
```

Comment:
Handle the `model is None` case separately here, so that we don't miss the `kwargs.pop('device')` when `model is None` (it used to be done inside `prep_model_and_emb_tf` when `backend == 'tensorflow' even if `model is None`). When we add pytorch support a bit more logic might be needed here to avoid pop'ing `device` when the embedding is a pytorch model...